examples/reshape.cpp

```diff
@@ -0,0 +1,162 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/Timer.hpp"
+
+
+#include "memoryManager.hpp"
+
+/*
+ *  RAJA Reshape method
+ *
+ *  This example will intialize array using
+ *  the RAJA Reshape method. The Reshape
+ *  method offers right and left most unit
+ *  stride.
+ *
+ */
+
+/*
+ * Define number of threads in a GPU thread block
+ */
+#if defined(RAJA_ENABLE_CUDA)
+constexpr int CUDA_BLOCK_SIZE = 256;
+#endif
+
+#if defined(RAJA_ENABLE_HIP)
+constexpr int HIP_BLOCK_SIZE = 256;
+#endif
+
+//
+//Function for checking results
+//
+void checkResult(int *ptr, int K, int N, int M);
+
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout << "\n\nRAJA reshape method example....\n"<< std::endl;
+
+  const int K = 3;
+  const int N = 1;
+  const int M = 2;
+
+  // Allocate memory for pointer
+  int *Rptr = memoryManager::allocate<int>(K * N * M);
+  int *Lptr = memoryManager::allocate<int>(K * N * M);
+  int *Cptr = memoryManager::allocate<int>(K * N * M);
+
+//----------------------------------------------------------------------------//
+//
+// Initialize memory using right most unit stride
+//
+//----------------------------------------------------------------------------//
+  std::cout << "\n\nInitialize array with right most indexing...\n";
```

Comment:
I think these implementations are much easier to understand if your intent is just to show the mechanics of View reshaping. 👍